serialize.go

```diff
@@ -28,95 +28,43 @@ const (
 	channelSize = 256 // 256 is chosen because 256*8 = 2kb, or about the cost of a go routing
 )
 
-type job struct {
-	f    func()
-	done chan struct{}
-}
-
 // Executor - a struct that can be used to guarantee exclusive, in order execution of functions.
 type Executor struct {
-	execCh    chan *job
-	init      sync.Once
-	count     int32
-	jobs      []*job
-	runningCh chan struct{}
-	idle      bool
+	orderCh     chan func()
+	buffer      []func()
+	bufferMutex sync.Mutex
+	init        sync.Once
+	count       int32
 }
 
 // AsyncExec - guarantees f() will be executed Exclusively and in the Order submitted.
 //        It immediately returns a channel that will be closed when f() has completed execution.
 func (e *Executor) AsyncExec(f func()) <-chan struct{} {
-	// Initialize *once*
 	e.init.Do(func() {
-		e.execCh = make(chan *job, channelSize)
+		e.orderCh = make(chan func(), channelSize)
 	})
-	exec := &job{
-		f:    f,
-		done: make(chan struct{}),
+	result := make(chan struct{})
+	e.orderCh <- func() {
```

Comment:
@edwarnicke Note: this channel should provide FIFO order of goroutines that called AsyncExec

Reply:
Agree, it will 👍